src/App/Fossa/EmbeddedBinary.hs

```diff
@@ -86,13 +88,22 @@ cleanupThemisBins (ThemisBins a b) = traverse_ cleanupExtractedBinaries [unTag a
 withThemisAndIndex :: Has (Lift IO) sig m => (ThemisBins -> m c) -> m c
 withThemisAndIndex = bracket extractThemisFiles cleanupThemisBins
 
+-- When running Themis we always need both the themis-cli and the decompressed index.gob
 extractThemisFiles :: Has (Lift IO) sig m => m ThemisBins
 extractThemisFiles = do
-  themisActual <- applyTag @ThemisBinary <$> extractEmbeddedBinary Themis
-  -- TODO: The gob is still xzipped, unzip it here
-  themisIndex <- applyTag @ThemisIndex <$> extractEmbeddedBinary ThemisIndex
-
-  pure $ ThemisBins themisActual themisIndex
+  themisBin <- extractEmbeddedBinary Themis
+  let themisActual = applyTag @ThemisBinary themisBin
+  -- decompress index.gob.gz binary and write it to disk
+  let bin = toStrict $ Lzma.decompress $ toLazy embeddedBinaryThemisIndex
```

Comment:
Instead of forcing this back to strict right away, we could use the lazy version of writeFile.  Generally, using lazy IO with lazy bytestrings is dangerous, but the scope dies right at the end of the function, since we run the IO completely with `sendIO`, so it's likely completely safe here.

If this works, it will probably be better in terms of memory usage, especially since the file is so large once unzipped.  In it's current form, it balloons up to the full size of the file, which is about 100MB, and then drains all at once.  With lazy bytestring, LZMA will chunk the bytestrings however it wants, on-demand (demand coming from the file writer).